csharp/private/common.bzl

```diff
@@ -72,12 +75,10 @@ def fill_in_missing_frameworks(providers):
 
         # There are at most 2 elements in FrameworkCompatibility[tfm], so this
         # nested loop isn't bad.
-        for compatible_tfm in FrameworkCompatibility[tfm]:
-            if compatible_tfm not in providers:
-                continue
-
+        # Order by providers that didn't "cross the netstandard boundary" so
+        # newer netstandard will be preferred, if applicable
+        for (base, compatible_tfm) in sorted([(providers[compatible_tfm], compatible_tfm) for compatible_tfm in FrameworkCompatibility[tfm] if compatible_tfm in providers], key=_get_provided_by_netstandard):
```

Comment:
No thought into line length / how to format python code. Feel free to commit fixing this (and 90)

Reply:
I've been keeping it to 80. I think buildifier can enforce this and we'll be able to run that in CI soon anyway so I'm not worried (we'll get to it)